zql/parser-support.go

```diff
@@ -114,3 +116,56 @@ func makeUnicodeChar(chars interface{}) string {
 
 	return string(r)
 }
+
+var syntaxRegexp = regexp.MustCompile(`^([0-9]+):([0-9]+) \(([0-9]+)\): no match found`)
+
+func ImproveError(src string, e error) error {
+	fmt.Printf("===\n%s===\n", src)
+	hits := syntaxRegexp.FindStringSubmatch(e.Error())
+	if len(hits) != 4 {
+		return e
+	}
+	lineNo, err := strconv.Atoi(hits[1])
+	if err != nil {
+		return e
+	}
+	colNo, err := strconv.Atoi(hits[2])
+	if err != nil {
+		return e
+	}
+	if colNo > 100 {
+		return e
+	}
+	_, err = strconv.Atoi(hits[3])
+	if err != nil {
+		return e
+	}
+	lineNo--
```

Comment:
```suggestion
	el, ok := e.(errList)
	if !ok || len(el) != 1 {
		return e
	}
	pe, ok := el[0].(*parserError)
	if !ok {
		return e
	}
	lineNo, colNo := pe.pos.line-1, pe.pos.col
```

Reply:
doh!  I went to do that and saw the error values weren't exported without thinking this code lives in the same package :-)